src/components/Calender/CalendarSelector.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { useState } from 'react';
+import { Calendar } from './Calendar';
+import { Icon, Layout, Text } from '@ui-kitten/components';
+
+import { View, StyleSheet } from 'react-native';
+import { months } from '../../services/Calendar/models';
+import { useStringDate } from '../../services/Hooks/useStringDate';
+import { selectSwapDate } from '../../store/actions';
+import { useDispatch, useSelector } from 'react-redux';
+import { TouchableOpacity } from 'react-native-gesture-handler';
+
+export const CalendarSelector = (props) => {
+    const [displayedDate, setDisplayedDate] = useState(props.displayedDate);
+    const [isLeap, year, month] = useStringDate(displayedDate);
+    const selectedDate = useSelector((state) => state.swapReducer.swapDate);
+    const dispatch = useDispatch();
+
+    const onDateTilePress = (date, data) => {
```

Comment:
not a fan of creating a function that does another function. also `data` argument isn't being used anywhere

Reply:
ted said he will need more stuff in here